processing/src/main/java/org/apache/druid/query/operator/window/WindowFrame.java

```diff
@@ -21,159 +21,226 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Lists;
-import org.apache.druid.query.operator.ColumnWithDirection;
+import org.apache.druid.annotations.SubclassesMustOverrideEqualsAndHashCode;
+
+import javax.annotation.Nullable;
 
-import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
-import java.util.stream.Collectors;
 
-public class WindowFrame
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes(value = {
+    @JsonSubTypes.Type(name = "unbounded", value = WindowFrame.Unbounded.class),
```

Comment:
is this really the right abstraction? "unbounded" isn't really a frame type, rather a frame start/end thing?

Why not just make a singleton instance for an 'unbounded' `Rows.class`? (assuming this is meant to be `ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING` or whatever)

Reply:
this is more like just syntactic sugar for native queries; yes it could also be `{type: rows }` - that's kinda the same - `unbounded` is just more explicit if its present in the native query.

Reply:
removed `unbounded` - it made things simpler as well!